File: backend/src/main/java/com/gxs/bank/repository/PromotionRepository.java
package com.gxs.bank.repository;

import com.gxs.bank.model.Promotion;
import org.springframework.data.jpa.repository.JpaRepository;
import org.springframework.stereotype.Repository;
import java.util.List;
import java.util.UUID;

@Repository
public interface PromotionRepository extends JpaRepository<Promotion, UUID> {
    List<Promotion> findByIsActiveTrueOrderByValidFromDesc();
}

File: backend-python/com/gxs/bank/repository/PromotionRepository.py
from com.gxs.bank.model.Promotion import Promotion
from com.gxs.bank.repository._base import BaseRepository


class PromotionRepository(BaseRepository):
    def __init__(self, db):
        super().__init__(db, Promotion)

    def findByIsActiveTrueOrderByValidFromDesc(self):
        return (
            self.db.query(Promotion)
            .filter(Promotion.isActive.is_(True))
            .order_by(Promotion.validFrom.desc())
            .all()
        )
